magento/magento2
# Fix: remove TestObserver class

### Description
Hello,

The `Magento\ProductAlert\Controller\Add\TestObserver`, as I can see, was used for testing propose. It can run methods for sending product alert (price and stock) emails and, also, error emails for admin: `Magento\ProductAlert\Model\Observer::process()`. The main reason why I created this PR and remove reported class is that it can be run by any logged in customer while `process()` method should be run only by an event or by cron.

Looking forward to your reply,
Alex

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios
1. Logged in to the frontend area using your customer account
2. Set the breakpoint for debugging in your IDE here: `Magento\ProductAlert\Controller\Add\TestObserver::execute()` line 18
2. Use next URL for test/debug: <base_url>/productalert/add/testObserver/
3. In browser you will see the blank page. In your debug console you will be stopped at the breakpoint and can continue with the `process()` method

### Contribution checklist
